src/lesson/lesson03.js

```diff
@@ -248,13 +248,15 @@ function Lesson03ExerciseStepPlayer(rows, cols, robot_position,
                                     robot_direction,
                                     components_positions,
                                     machines_positions,
-                                    goal) {
+                                    goal,
+                                    hidden_positions) {
```

Comment:
In the steps, all the hidden positions come before the goal, if you could change the order of these two parameters would be easier than change all the steps.